swarm_test.go

```diff
@@ -215,6 +219,20 @@ func TestNewSwarmFailure(t *testing.T) {
 			configure: func(config *api.Config) {
 				config.SwapAPI = ""
 				config.SwapEnabled = true
+				config.NetworkID = swap.AllowedNetworkID
+			},
+			check: func(t *testing.T, s *Swarm, _ *api.Config) {
+				if s != nil {
+					t.Error("swarm struct is not nil")
+				}
+			},
+		},
+		{
+			name: "with swap enabled and default network ID",
+			configure: func(config *api.Config) {
+				config.SwapAPI = ""
```

Comment:
The correct combination here (or at least one other possible combination) I think should be that `SwapAPI ` is set, but also the `DefaultNetworkID`. Otherwise it could actually fail because the API is not set (even if right now of course, you first check for the network ID, which is good so, but let's assume someone changes it...). Nitpicking I know :)

Reply:
You're totally right! In fact, I meant to do exactly this, but it seems it slipped my mind :)

Thanks for the catch! Fixed.